### ISSUE TITLE
CRITICAL:root:Cannot load caffe2.python. Error: DLL load failed: The specified module could not be found.

### ISSUE THREAD

**Issue by 137996047** [pytorch/pytorch#22859]
## ❓ Questions and Help
I'm installing into a conda env. it is on Windows 10 . No GPU on machine.
I am following instructions from https://caffe2.ai/docs/getting-started.html?platform=windows&configuration=compile

And the build is success.
![image](https://user-images.githubusercontent.com/30762967/61205807-9eb70800-a723-11e9-8cb1-375938b085a1.png)

Then I copy C:\Users\v-hadong\pytorch\build\caffe2\python\caffe2_pybind11_state.pyd to Python’s DLL folder C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\DLLs\

and then the error is happened as the following picture:
![image](https://user-images.githubusercontent.com/30762967/61207459-bb553f00-a727-11e9-9f1d-e3968dfcae70.png)

**Comment by peterjc123** [comment]
You need to copy the DLLs generated by LibTorch in the lib directory, too.

**Comment by 137996047** [comment]
> directory
Thank you for your replay, I copy  `C:\Users\v-hadong\pytorch\build\lib\caffe2_pybind11_state.lib`  to `C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\libs`. 
But it doesn't work.

**Comment by peterjc123** [comment]
No, I mean something like `torch.dll`, `c10.dll`.

**Comment by 137996047** [comment]
> No, I mean something like `torch.dll`, `c10.dll`.

 I copy all *.dll to `C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\libs` ,
 it doesn't work too.

**Comment by 137996047** [comment]
> No, I mean something like `torch.dll`, `c10.dll`.

When I run the build.bat, I `set BUILD_SHARED_LIBS=ON`.  Does this matter?

**Comment by peterjc123** [comment]
Yes, that means the DLLs will be used instead of a complete executable without dependencies.

**Comment by peterjc123** [comment]
You should copy them to `C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\DLLs`, not `libs`.

**Comment by 137996047** [comment]
> You should copy them to `C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\DLLs`, not `libs`.

I copy torch.dll, c10.dll to  `C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\DLLs`, it doesn't work too.

And I use the dependencies analyse the `caffe2_pybind11_state.pyd`, it shows:
![image](https://user-images.githubusercontent.com/30762967/61262911-a7f2b400-a7b9-11e9-8c5b-af6f7f70f0b2.png)

**Comment by peterjc123** [comment]
Would you please add `C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\DLLs` into the environment variable `PATH` and try again?

**Comment by 137996047** [comment]
> Would you please add `C:\Users\v-hadong\AppData\Local\Continuum\anaconda2\envs\caffe2\DLLs` into the environment variable `PATH` and try again?

I try do this, it doesn't work too.
And I found python 2.7 dependency is in base env not in the caffe2.
![image](https://user-images.githubusercontent.com/30762967/61265981-5d773480-a7c5-11e9-97d4-cd3cef53fac3.png)

Then I do the same procedures as above in the base env, but it doesn't work too.

**Comment by peterjc123** [comment]
You'll need to reopen a new CMD window to take that into effect.

**Comment by 137996047** [comment]
> You'll need to reopen a new CMD window to take that into effect.

I try it, but doesn't work.

**Comment by peterjc123** [comment]
I've introduced some methods for dealing with dll loading problems in this post: https://github.com/pytorch/pytorch/issues/20408. You can have a try.
